src/mon/MonCommands.h

```diff
@@ -1505,3 +1503,7 @@ COMMAND_WITH_FLAG("dump_historic_slow_ops",
             "show recent slow ops",
             "mon", "r",
             FLAG(TELL))
+COMMAND_WITH_FLAG("dump_messenger",
```

Comment:
```suggestion
COMMAND_WITH_FLAG("messenger dump",
```

(consistency)

Reply:
Is there a reason you didn't make this command handler part of the Messenger? It should probably be `MessengerHook` (grep `MDSSocketHook`) which is attached to the `CephContext::get_admin_socket()` when the Messenger instance is created. Then you don't need to define the command for each component.

Reply:
The last push replaces the individual "dump messenger" commands with "messenger dump <name>". All registered on messenger start.

In general I much prefer the single implementation. The drawback is that tools now need to discover available messengers and get data for each individually. 

I updated the cntop tool to call "help", grep the messenger name from the commands and then call messenger dump for each. It works, but feels a bit hacky.

Is there a precedence for dynamic asok commands and subcommand discovery that I could use as inspiration?